### ISSUE TITLE
Capacitor UI not showing after update from 0.4.3 to 0.4.4

### ISSUE THREAD

**Issue by eduardowitter** [gimlet-io/capacitor#124]
[capacitor-645fdf6747-k8cx8_capacitor.log](https://github.com/user-attachments/files/17216371/capacitor-645fdf6747-k8cx8_capacitor.log)

After the update from 0.4.3 to 0.4.4 , the Capacitor UI is not displaying any resources:

![image](https://github.com/user-attachments/assets/a29d3936-f47c-442b-a70a-003a466daa4e)

Capacitor logs are attached.

After rolling back to 0.4.3 I can get the UI working again, so I'm assuming this is related to 0.4.4.

**Comment by hugginsio** [comment]
I am having the same issue with identical log messages on Kubernetes v1.29.3 via Orbstack.

**Comment by ivaylo-mi** [comment]
I have the same issue and added the following to the ServiceAccount
````
- apiGroups:
  - infra.contrib.fluxcd.io
  resources:
  - terraforms
  verbs:
  - get
  - watch
  - list
````
But it didn't help, only the error messages changed to
`level=warning msg="could not marshal event: the server could not find the requested resource"`

**Comment by SvenW** [comment]
Sorry for that. Should be fixed here https://github.com/gimlet-io/capacitor/pull/125

**Comment by SvenW** [comment]
But 

> I have the same issue and added the following to the ServiceAccount
> 
> ```
> - apiGroups:
>   - infra.contrib.fluxcd.io
>   resources:
>   - terraforms
>   verbs:
>   - get
>   - watch
>   - list
> ```
> 
> But it didn't help, only the error messages changed to `level=warning msg="could not marshal event: the server could not find the requested resource"`

The terraform controller needs to be opt-in in order to get it working. So if you're not using that it will break at the moment

**Comment by laszlocph** [comment]
v0.4.5 has the fix

`flux reconcile source oci capacitor` should pull it right in.

**Comment by FelipeEmerimRandon** [comment]
We are having this issue in v0.4.8, with the same logs:

    time="2024-11-22T13:09:22Z" level=warning msg="could not marshal event: terraforms.infra.contrib.fluxcd.io is forbidden: User \"system:serviceaccount:flux-system:capacitor\" cannot list resource \"terraforms\" in API group \"infra.contrib.fluxcd.io\" at the cluster scope"

**Comment by evanmoelter** [comment]
I also had this issue in v0.4.8. I believe part of the issue was me using a different helm chart  than the official one which was missing the RBAC to access the terraform objects. Once I added the `role` from the official helm chart and an appropriate `rolebinding`, this issue was resolved. (For reference, [this is the related PR](https://github.com/evanmoelter/hcc/pull/122/files) in my home cluster setup.)
